columnflow/columnar_util.py

```diff
@@ -1032,23 +1039,7 @@ def layout_ak_array(data_array: np.array | ak.Array, layout_array: ak.Array) ->
         c = layout_ak_array(a, b)
         # <Array [[], [1.0, 2.0], [], [3.0, 4.0, 5.0]] type='4 * var * float32'>
     """
-    # flatten and convert to content
-    if isinstance(data_array, np.ndarray):
-        data = ak.contents.NumpyArray(np.reshape(data_array, (-1,)))
-    elif isinstance(data_array, ak.Array):
-        data = ak.flatten(data_array, axis=None)  # might create a copy
-    else:
-        raise TypeError(f"unhandled type of data array {data_array}")
-
-    # infer the offsets
-    if getattr(layout_array, "layout", None) is None:
-        raise ValueError(f"layout array {layout_array} does not have a valid layout")
-    if getattr(layout_array.layout, "offsets", None):
-        offsets = layout_array.layout.offsets
-    else:
-        raise ValueError(f"offsets extraction not implemented for layout array {layout_array}")
-
-    return ak.Array(ak.contents.ListOffsetArray(offsets, data))
+    return ak.unflatten(data_array, ak.num(layout_array))
```

Comment:
This simplifies things a lot, +1 for using `unflatten` here, which should give the same results as the previous implementation for arrays with a single jagged dimension.

On that note, now that I am reading the docstring again, I think the phrase "arbitrarily ragged" is a bit misleading, as it seems to imply that this works at deeper nesting levels as well. A simple fix would be to just remove the word "arbitrarily".

As a minor addition, I would suggest to specify the axis along which to take the counts/unflatten explicitly, so that the intention is clear. Using the `awkward` defaults, that would be:
```suggestion
    return ak.unflatten(data_array, ak.num(layout_array, axis=1), axis=0)
```

Reply:
Mh, I think I overlooked the *detail* that unflatten only accepts a 1-dimension array of counts. This is quite limiting in my opinion. In this case I would be inclined to keep the current implementation.

Reply:
Hm, I think we should look into this in more detail. I have the impression that the old implementation does not produce the expected results if the `layout_array` has multiple nesting levels. It just lays out the flattened `data_array` according to the first jagged level, but does not thread beyond that. It also does not check that the flattened `data_array` fits into the `layout_array` exactly, which is important here IMO.

Do we actually have a use case with multiple nesting levels? If so, we can use it for testing and prototyping a new implementation, but for now I would actually stick with simply-jagged layouts and `ak.unflatten` here.

Reply:
Ok, I agree. I just tested the old implementation for arrays with more than 2 dimensions and it indeed fails. I have to wrap my head around what the "layout" object actually is, but in the meantime, let's stick to the shorter implementation then.

(btw, the old one also failed when `data_array` was not a numpy but an awkward array, since the return type of ak.flatten(), a usualy array, is not a content-subtype).

Reply:
I added your suggestions :+1:

Reply:
OK, thanks! Then let's keep this as it is and revisit the issue if/when a use-case for deep-nested layouting comes up.